src/dataset-builder/ConceptSearch.test.ts

```diff
@@ -0,0 +1,187 @@
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import _ from 'lodash/fp';
+import { act } from 'react-dom/test-utils';
+import { h } from 'react-hyperscript-helpers';
+import { ConceptSearch } from 'src/dataset-builder/ConceptSearch';
+import { dummyDatasetModel, dummyGetConceptForId } from 'src/dataset-builder/TestConstants';
+import { DataRepo, DataRepoContract, SnapshotBuilderConcept } from 'src/libs/ajax/DataRepo';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+
+// This is necessary to avoid waiting for the delayed (debounced) input change event.
+jest.mock('src/components/input', () => {
+  return {
+    ...jest.requireActual('src/components/input'),
+    withDebouncedChange: (component) => component,
+  };
+});
+
+jest.mock('src/libs/ajax/GoogleStorage');
+type DataRepoExports = typeof import('src/libs/ajax/DataRepo');
+jest.mock('src/libs/ajax/DataRepo', (): DataRepoExports => {
+  return {
+    ...jest.requireActual('src/libs/ajax/DataRepo'),
+    DataRepo: jest.fn(),
+  };
+});
+describe('ConceptSearch', () => {
+  const onCancel = jest.fn();
+  const onCommit = jest.fn();
+  const onOpenHierarchy = jest.fn();
+  const actionText = 'action text';
+  const datasetId = '0';
+  const domainOption = dummyDatasetModel()!.snapshotBuilderSettings!.domainOptions[0];
+
+  const renderSearch = (initialSearch = '', initialCart: SnapshotBuilderConcept[] = []) =>
+    render(
+      h(ConceptSearch, {
+        actionText,
+        initialSearch,
+        initialCart,
+        onCancel,
+        onCommit,
+        onOpenHierarchy,
+        datasetId,
+        domainOption,
+      })
+    );
+
+  const displayedConcepts = [dummyGetConceptForId(102), dummyGetConceptForId(103)];
+  const mockSearch = jest.fn();
+
+  const mockDataRepoContract: DataRepoContract = {
+    dataset: (_datasetId) =>
+      ({
+        searchConcepts: mockSearch,
+      } as Partial<DataRepoContract['dataset']>),
+  } as Partial<DataRepoContract> as DataRepoContract;
+  asMockedFn(DataRepo).mockImplementation(() => mockDataRepoContract as DataRepoContract);
+  mockSearch.mockResolvedValue({
+    result: displayedConcepts,
+  });
+
+  it('renders the concept search with an initial search result', async () => {
```

Comment:
How would you feel about `'renders the concept search with an empty initial search'`? Do you think that there's an overlap with the current name and a non-empty initial search?

Reply:
I can see that the description could be read a few different ways. What I meant is that the page is displayed with the initial result, which is all elements. (Actually all results are all elements since no filtering actually occurs in the test code.)

So the page content isn't affected by the search string itself; the only thing that's affected is if the ajax mock is called or not.